File: _submissions/18UEI056_Prashun/Day1/subset.cpp
void backtrack(vector<int>&A,vector<int>&temp,vector<vector<int>>&ans,int index)
{ ans.push_back(temp);
for(int i=index;i<A.size();i++)
{ temp.push_back(A[i]);
    backtrack(A,temp,ans,i+1);
    temp.pop_back();
}
}

vector<vector<int> > Solution::subsets(vector<int> &A) {
    vector<int>temp;
    vector<vector<int>>ans;
    sort(A.begin(),A.end());
    backtrack(A,temp,ans,0);
    return ans;
}

File: _submissions/19UCS159_Tharun/Day1/subSet.java
public class Solution {
  public int[][] subsets(int[] A) {
    Arrays.sort(A);
    ArrayList<int[]> res = new ArrayList<>();
    int n = A.length;
    solve(A, 0, new ArrayList<>(), res);
    int[][] ans = new int[res.size()][];
    for (int i = 0; i < res.size(); i++) {
      ans[i] = res.get(i);
    }
    return ans;
  }

  private void solve(int[] A, int i, ArrayList<Integer> curr, ArrayList<int[]> res) {
    int[] cur = new int[curr.size()];
    for (int k = 0; k < curr.size(); k++) {
      cur[k] = curr.get(k);
    }
    res.add(cur);
    if (i == A.length) {
      return;
    }
    for (int j = i; j < A.length; j++) {
      ArrayList<Integer> newCurr = new ArrayList<>(curr);
      newCurr.add(A[j]);
      solve(A, j + 1, newCurr, res);
    }

  }
}
